grazer/src/main/java/edu/asu/diging/grazer/core/domain/ITransformationFilesMetadata.java

```diff
@@ -0,0 +1,38 @@
+package edu.asu.diging.grazer.core.domain;
+
+import java.time.OffsetDateTime;
+
+import edu.asu.diging.grazer.core.domain.impl.TransformationFilesImpl;
+
+/**
+ * This class stores all the details about a transformation file and pattern file 
+ * i.e their IDs, labels, descriptions, file names, data uploaded, and uploader.
+ * 
+ * @author mshah18
+ *
+ */
+public interface ITransformationFilesMetadata {
+
+    int getId();
+    
+    String getLabel();
+
+    void setLabel(String label);
+
+    String getDescription();
+
+    void setDescription(String description);
+    
+    TransformationFilesImpl getFiles();
+    
+    void setFiles(TransformationFilesImpl files);
```

Comment:
you should use the interface here

Reply:
Using the interface threw a nullvalueinnestedpathexception while uploading files in the jsp page and so I changed it to use the actual class instead

Reply:
now that you are not using that anymore in the jsp page, those should be interfaces.